packages/data-context/src/sources/migration/codegen.ts

```diff
@@ -55,7 +36,7 @@ export interface CreateConfigOptions {
   hasTypescript: boolean
 }
 
-export async function createConfigString (cfg: OldCypressConfig, options: CreateConfigOptions) {
+export async function createConfigString (cfg: Partial<Cypress.Config>, options: CreateConfigOptions) {
```

Comment:
Why did you change this type?

`integrationFolder` & `componentFolder` should not work with `Cypress.Config`...

Although maybe they do and that is a separate issue

Reply:
The type `OldCypressConfig` is wrong - we receiving the legacy config here as this arg, fully including `integrationFolder`, even if it's not included in the type. I'm not sure why this type was created in the first place?

`integrationFolder` etc get removed later in `reduceConfig` - they aren't included in the created `cypress.config.js`.